tei_adapter/tei_id_extractor/src/main/scala/weco/catalogue/tei/id_extractor/GitHubBlobContentReader.scala

```diff
@@ -0,0 +1,70 @@
+package weco.catalogue.tei.id_extractor
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+
+import java.net.URI
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import io.circe.Decoder
+import uk.ac.wellcome.json.JsonUtil._
+import weco.http.client.HttpClient
+
+class GitHubBlobContentReader(httpClient: HttpClient, token: String)(
+  implicit ac: ActorSystem) {
+  implicit val ec = ac.dispatcher
+  def getBlob(uri: URI): Future[String] = {
+    val request = HttpRequest(
```

Comment:
I wonder if this would benefit from the HttpClient pattern @alexwlchan abstracted recently (https://github.com/wellcomecollection/scala-libs/blob/main/http/src/main/scala/weco/http/client/HttpClient.scala) for reducing WireMock interactions?

Reply:
There's a conversation with @alexwlchan above about this. I don't want to bypass wiremock cause that requires me understanding properly the response and converting in an HttpResponse object accurately, and I don't trust myself on this, I'd rather delegate it to wiremock